daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DataProcessor.scala

```diff
@@ -311,18 +271,13 @@ class DataProcessor private (
     val oos = new ObjectOutputStream(new GZIPOutputStream(os))
 
     //
-    // Make a copy of this object, so that our state mods below don't side-effect the user's object.
-    // Saving shouldn't have side-effects on the state of the object.
-    //
-    //
-    // Note that the serialization system *does* preserve these two settings. This is for general serialization
-    // that may be required by other software (e.g., Apache Spark)
+    // Make a copy of this object so that we can make its saved state
+    // different than its original state.  Note that the serialization
+    // system *does* preserve validationMode since it may be required by
+    // other software like Apache Spark. But for our save/reload purposes,
+    // we don't want validationMode preserved.
     //
-    // But for our save/reload purposes, we don't want them preserved.
-    //
-
     val dpToSave = this.copy(
-      externalVars = Queue.empty[Binding], // explicitly set these to empty so restored processor won't have them.
```

Comment:
I think we might still want this line if we add back the VariableMap to the SerializableDataProcessor. Or i guess really we want this to be `externalVars = ssrd.originalValues`?

Reply:
I think you meant `variableMap = ssrd.originalVariables` since we've gotten rid of all uses/mentions of `externalVars`.  Anyway, should a saved processor preserve its own `variableMap` or replace it with `ssrd.originalVariables`?  That is the real question we need to ask.  Mike @mbeckerle , do you want to weigh in?

Reply:
I ended up omitting variableMap because variableMap already has a default value (variableMap.copy()) which is copied if not passed explicitly.  

Mike @mbeckerle , will you please review all the changes so far?  I want to get this PR done and out of the way.

Reply:
Doesn't this change the behavior? I believe previously if you `save()` and `reload()` you would get the original variables values defined in the schema? With this, doesn't a `reload()` get whatever variables were set on the `DataProcessor` at the time of `save()`?

Reply:
From Mikes other comment:

> Saving the parser does NOT lock-down/freeze or even utilize any variable values. Saving the parser is supposed to be transparent, i.e., only faster, than using the uncompiled schema.

So it seems like we do want `variableMap = ssrd.originalVariables` so that calling `save()` essentially erases any calls to `withVariableMap`

Reply:
Ah. Yes, steve is right.  We do want the originalVariables.